js/app.js

```diff
@@ -1,5 +1,40 @@
-angular.module('GiphyApp', [])
+angular.module('GiphyApp', ['infinite-scroll'])
 
-.run(function() {
-  console.log('App has loaded!');
-});
+.controller('GiphyCtrl', ['$scope', '$http', function($scope, $http) {
+  $scope.search = '';
+  $scope.searchGiphy = searchGiphy;
+
+function searchGiphy() {
+  console.log('success!')
+  $http.get('http://api.giphy.com/v1/gifs/search', {
+    params: {
+      q: $scope.search,
+      api_key: 'dc6zaTOxFJmzC'
+    }
+  })
+  .then(function success(res) {
+    // $scope.search = '';
+    $scope.giphy = res.data.data;
+    console.log($scope.giphy);
+  }, function error(res) {
+    console.log(res);
+  });
+}
+
+  // $scope.$watch = ('search', function(newValue, oldValue) {
+  //   $http.get('http://api.giphy.com/v1/gifs/search', {
+  //     params: {
+  //       q: $scope.search,
+  //       api_key: 'dc6zaTOxFJmzC'
+  //     }
+  //   }).then(function success(res) {
+  //     $scope.giphy = res.data.data;
+  //   }, function error(res) {
+  //     console.log(res);
+  //   })
+  // });
+
```

Comment:
Here is some code from my deliverable to handle the infinite scrolling:

First I added the angular module 'infinite-scroll' (https://sroze.github.io/ngInfiniteScroll/):

    var giphyApp = angular.module('GiphyApp', ['infinite-scroll']);

And then the JS:

    $scope.search = function () {
        if ($scope.searchTerm != $scope.oldSearchTerm) {$scope.giphys = [];};
        var req = {
            url: "http://api.giphy.com/v1/gifs/search",
            method: 'GET',
            params: {
                api_key: $scope.api,
                offset: $scope.offset,
                q: $scope.searchTerm,
            }
        }
        $http(req).then(function success(res) {
            $scope.giphys = $scope.giphys.concat(res.data.data);
        }, function error (res) {
            console.log(res);
        });
    };

    $scope.myPagingFunction = function (){
        $scope.oldSearchTerm = $scope.searchTerm;
        $scope.offset = 25 * $scope.count;
        $scope.search();
        $scope.count++;
    };